test/facets/gas-reports.spec.ts

```diff
@@ -0,0 +1,411 @@
+
+  const secret = Buffer.from(privateKey);
+  const shards: Buffer[] = sss.split(secret, arg);
+
+  // const recovered = sss.combine(shards.slice(0, 2));
+  // console.log(recovered.toString());
+
+  const sarcoName = `Init sarco (${arg.shares})`;
+  const sarcoId = ethers.utils.solidityKeccak256(["string"], [sarcoName]);
+
+  const resurrectionTimeInFuture =
+    (await time.latest()) + time.duration.weeks(1);
+
+  const [archaeologists, signatures] = await _spawnArchaologistsWithSignatures(
+    shards,
+    sarcoId
+  );
+
+  await setupArchaeologists(
+    archaeologistFacet,
+    archaeologists.map((a) => a.signer),
+    diamondAddress,
+    embalmer,
+    sarcoToken
+  );
+
+  // Choose arweave archaeologist.
+  arweaveArchaeologist = archaeologists[0].signer;
+
+  const tx = await _initializeSarcophagus(
+    sarcoName,
+    BigNumber.from(resurrectionTimeInFuture),
+    sarcoId,
+    archaeologists,
+    arweaveArchaeologist.address,
+    arg.threshold
+  );
+
+  tx.wait();
+
+  const arweaveSignature = await sign(
+    arweaveArchaeologist,
+    "arweaveTxId",
+    "string"
+  );
+
+  const finTx = await embalmerFacet.finalizeSarcophagus(
+    sarcoId,
+    signatures.slice(1, signatures.length), // first signer is arweave archaeologist. Exclude their signature
+    arweaveSignature,
+    "arweaveTxId"
+  );
+  finTx.wait();
+
+  // check shard lengths
+  expect(shards[0].length).to.eq(shards[1].length).to.eq(1058);
+
+  // check hashed shard lengths
+  expect(archaeologists[0].hashedShard.length)
+    .to.eq(archaeologists[0].hashedShard.length)
+    .to.eq(66);
+
+  return { sarcoId, archaeologists, arweaveSignature };
+}
+
+async function _runRewrapTest(sarcoId: string) {
+  // Define a new resurrection time one week in the future
+  const newResurrectionTime = (await time.latest()) + time.duration.weeks(1);
+
+  await embalmerFacet
+    .connect(embalmer)
+    .rewrapSarcophagus(sarcoId, newResurrectionTime);
+}
+
+async function _runUnwwrapTest(
+  sarcoId: string,
+  archaeologists: TestArchaeologist[]
+) {
+  await time.increase(time.duration.weeks(1));
+
+  for await (const arch of archaeologists) {
+    await archaeologistFacet
+      .connect(arch.signer)
+      .unwrapSarcophagus(sarcoId, arch.unencryptedShard);
+  }
+}
+
+async function _spawnArchaologistsWithSignatures(
+  shards: Buffer[],
+  sarcoId: string
+): Promise<[TestArchaeologist[], SignatureWithAccount[]]> {
+  const archs: TestArchaeologist[] = [];
+  const signatures: SignatureWithAccount[] = [];
+
+  for (let i = 0; i < shards.length; i++) {
+    const acc = signers[i + 2];
+
+    // todo: confirm this data is correct
+    const signature = await sign(acc, sarcoId, "bytes32");
+    // const pubKey = utils.recoverPublicKey(
+    //   ethers.utils.hashMessage(sarcoId),
+    //   signature
+    // );
+
+    // const encryptedShardBuffer = await encrypt(_hexToBytes(pubKey), shards[i]);
```

Comment:
@DarksightKellar -- what is this TODO and commented code for? Was this test code to confirm that the signature is valid?

Reply:
I've removed these bits in my formatting branch